app/src/main/java/org/wikipedia/descriptions/DescriptionEditView.kt

```diff
@@ -0,0 +1,371 @@
+        binding.viewDescriptionEditTextLayout.setErrorIconTintList(colorStateList)
+        binding.viewDescriptionEditTextLayout.setErrorTextColor(colorStateList)
+        binding.viewDescriptionEditTextLayout.boxStrokeErrorColor = colorStateList
+        layoutErrorState(text)
+    }
+
+    private fun clearError() {
+        binding.viewDescriptionEditTextLayout.error = null
+    }
+
+    private fun layoutErrorState(text: CharSequence?) {
+        // explicitly clear the error, to prevent a glitch in the Material library.
+        clearError()
+        binding.viewDescriptionEditTextLayout.error = text
+        if (!text.isNullOrEmpty()) {
+            post {
+                if (isAttachedToWindow) {
+                    binding.viewDescriptionEditScrollview.fullScroll(FOCUS_DOWN)
+                }
+            }
+        }
+    }
+
+    private fun enqueueValidateText() {
+        removeCallbacks(textValidateRunnable)
+        postDelayed(textValidateRunnable, TEXT_VALIDATE_DELAY_MILLIS.toLong())
+    }
+
+    private fun validateText() {
+        isTextValid = true
+        val text = binding.viewDescriptionEditText.text.toString().toLowerCase(Locale.getDefault()).trim()
+        if (text.isEmpty()) {
+            isTextValid = false
+            clearError()
+        } else if (text.length < 2) {
+            isTextValid = false
+            setError(context.getString(R.string.description_too_short))
+        } else if ((action == DescriptionEditActivity.Action.ADD_DESCRIPTION || action == DescriptionEditActivity.Action.TRANSLATE_DESCRIPTION) &&
+                (listOf(".", ",", "!", "?").filter { text.endsWith(it) }).isNotEmpty()) {
+            isTextValid = false
+            setError(context.getString(R.string.description_ends_with_punctuation))
+        } else if ((action == DescriptionEditActivity.Action.ADD_DESCRIPTION || action == DescriptionEditActivity.Action.TRANSLATE_DESCRIPTION) &&
+                LanguageUtil.startsWithArticle(text, pageTitle.wikiSite.languageCode())) {
+            setWarning(context.getString(R.string.description_starts_with_article))
+        } else if ((action == DescriptionEditActivity.Action.ADD_DESCRIPTION || action == DescriptionEditActivity.Action.TRANSLATE_DESCRIPTION) &&
+                pageTitle.wikiSite.languageCode() == "en" && Character.isUpperCase(binding.viewDescriptionEditText.text.toString()[0])) {
+            setWarning(context.getString(R.string.description_starts_with_uppercase))
+        } else if (isLanguageWrong) {
+            setWarning(context.getString(R.string.description_is_in_different_language,
+                    WikipediaApp.getInstance().language().getAppLanguageLocalizedName(pageSummaryForEdit.lang)))
+        } else {
+            clearError()
+        }
+        updateSaveButtonEnabled()
+    }
+
+    fun setHighlightText(text: String?) {
+        if (text != null && originalDescription != null) {
+            val scrollDelayMs = 500
```

Comment:
Would it be better if change to `val scrollDelayMs = 500L` and reduce `toLong()` ?